### PR TITLE
fix(slack): preserve completion section integrity

### PR DESCRIPTION
**Pull request by ColeMurray**
## Summary

Follow-up to #1147.

- preserve completion whitespace exactly and avoid splitting Unicode surrogate pairs at section or truncation boundaries
- recognize fenced spans that open and close on the same line, while retaining language metadata for continued fences
- stop section generation as soon as the 20-section response budget is exhausted
- omit top-level completion text so Slack derives screen-reader text from the full block content

## Root cause

The sectioner normalized paragraph separators, sliced UTF-16 strings at arbitrary code-unit offsets, and tracked at most one fence delimiter per line. It also generated the entire section list before truncating it. Completion delivery continued to send a 150-character top-level fallback, so assistive clients did not receive the expanded block content.

## TDD evidence

Each behavioral fix began with a regression that failed against merged `main`:

- same-line fenced content produced an odd fence count
- repeated blank lines and inter-section separators were not reconstructed exactly
- a boundary after 2,999 ASCII characters split an emoji into lone surrogates
- truncation-marker insertion could also cut through a surrogate pair
- the completion request still contained top-level `text` instead of allowing Slack to derive it from blocks

The existing block-budget regression remains green while generation now exits when the next section would exceed the configured budget.

## Validation

- `npm test -w @open-inspect/shared` — 500 passed
- `npm test -w @open-inspect/slack-bot` — 374 passed
- `npm run typecheck` — all workspaces passed
- shared and Slack-bot lint passed
- shared and Slack-bot builds passed
- Prettier and `git diff --check` passed

## Summary by CodeRabbit

- **New Features**
  - Added a blocks-only Slack message helper, with optional thread context, enabling messages to be sent without a separate top-level text field.

- **Bug Fixes**
  - Improved long-response splitting to preserve whitespace, keep code fences balanced, and avoid breaking Unicode characters.
  - Made truncation more Unicode-safe, including insertion of the “...truncated” marker and safer budget handling.

- **Tests**
  - Added regression tests for section splitting, Unicode boundary behavior, code-fence correctness, truncation edge cases, and blocks-only payload structure.
  - Added coverage for the new blocks-posting helper and completion delivery payloads.

### CODE REVIEW

File: packages/slack-bot/src/completion/blocks.ts [hunk 1]
```diff
@@ -265,35 +286,40 @@ export function splitIntoSlackSections(
...
+        }
+      }
     }
+    return true;
   };
 
-  for (const paragraph of trimmed.split(/\n{2,}/)) {
-    if (paragraph.length <= maxChars) {
-      appendToken(paragraph, "\n\n");
+  sourceTokens: for (const sourceToken of trimmed.split(/(\n{2,})/)) {
```
[the remark this answers is not in the document]
Comment (ColeMurray) [reply packages/slack-bot/src/completion/blocks.ts]: Good catch. This was valid: we used the trimmed value for both emptiness classification and tokenization, which removed non-empty leading and trailing whitespace. Fixed in `ca7bb7b7` by using `text.trim()` only for the empty-input check and sectioning the original `text`. I also extended the losslessness regression to cover leading indentation/newlines and trailing spaces/newlines. All 374 Slack-bot tests, typecheck, and lint pass.

File: packages/shared/src/slack/client.ts [hunk 2]
```diff
@@ -183,7 +183,7 @@ export async function verifySlackSignature(
 export function postMessage(
   token: string,
   channel: string,
-  text: string,
+  text: string | undefined,
```
[the remark this answers is not in the document]
Comment (ColeMurray) [reply packages/shared/src/slack/client.ts]: Agreed. Allowing `undefined` weakened the shared contract and made a content-less request representable. Fixed in `82558792` by restoring required text on `postMessage` and adding an explicit `postBlocks` API that requires blocks. Completion delivery now uses that API. A shared-client regression verifies that the block-only request includes blocks and thread metadata while omitting top-level `text`. All shared and Slack-bot tests, repository-wide typechecking, lint, and build pass.

File: packages/slack-bot/src/completion/blocks.ts [hunk 3]
```diff
@@ -194,6 +194,22 @@ function closeSuffix(state: FenceState): string {
...
+  if (
+    end > 1 &&
+    lastCodeUnit >= 0xd800 &&
+    lastCodeUnit <= 0xdbff &&
+    nextCodeUnit >= 0xdc00 &&
+    nextCodeUnit <= 0xdfff
+  ) {
+    end -= 1;
+  }
+  return text.slice(0, end);
```
[the remark this answers is not in the document]
Comment (ColeMurray) [reply packages/slack-bot/src/completion/blocks.ts:211]: Good catch. Reproduced: a one-code-unit budget split `😀` into separate high- and low-surrogate sections. Fixed in `741efb77` by allowing the boundary helper to return an empty cut when the next complete code point cannot fit, then detecting that no-progress condition before iteration and throwing a clear `RangeError`. The regression covers `splitIntoSlackSections("😀", 1)`. All 375 Slack-bot tests, repository-wide typechecking, lint, formatting, and build pass.
